src/pages/dashboard/index.tsx

```diff
@@ -1,15 +1,89 @@
-import React, {useState} from "react";
+import React, { useState } from "react";
+// import React from "react";
 import { Grid } from "@material-ui/core";
-// import dashboard from "../../api/dashboard";
+import FilterInput from "../../components/FilterInput";
+import List from "../../components/List";
+import { data } from "../../api/dashboard"; // available devices
 
 export const DashboardPage = () => {
-  const [data] = useState(null);
+  const [availableDevices, setAvailableDevices] =
+    useState(data) || useState([]);
+  const [myDevices, setMyDevices] = useState([]);
+  const buttonText = "Add to My List";
+  const allDevices = data;
+
+  function handleAvailableListChange(index) {
+    // create a copy of our arrays
+    const availableDeviceList = [...availableDevices];
+    const myDeviceList = [...myDevices];
+
+    // remove the clicked item from the array
+    const removedItem = availableDeviceList.splice(index, 1);
+    // add the clicked item to the other array
+    const myDeviceListArrayNEW = myDeviceList.concat(removedItem);
+
+    // set our new state
+    setAvailableDevices(availableDeviceList);
+    setMyDevices(myDeviceListArrayNEW);
+  }
+
+  function handleMyListChange(index) {
+    // create a copy of our arrays
+    const myDeviceList = [...myDevices];
+    const availableDeviceList = [...availableDevices];
+
+    // remove the clicked item from the array
+    const removedItem = myDeviceList.splice(index, 1);
+    // add the clicked item to the other array
+    const availableDeviceListNEW = availableDeviceList.concat(removedItem);
+
+    // set our new state
+    setMyDevices(myDeviceList);
+    setAvailableDevices(availableDeviceListNEW);
+  }
+
+  function handleFilterListInput(value) {
+    let filteringArray = [];
+
+    // generate our new filtered list from allDevices
+    // compare each item against our passed in filtering value
+    allDevices.map((device) => {
+      // grab all of the array values and convert to comma separated sting
+      const deviceValues = Object.values(device).toString().toLowerCase();
```

Comment:
@jenny-k This is still a little off from the requirement, looks like it's searching a string composed of all the field values of the entire object.

Requirement: The Available Device List must be filterable by typing text into an input field. The filter must search the Manufacturer, Model, and Platform fields.

Suggestion: Do an extra map on the device and pull just those 3 fields into a new object, then do your Object.values() call on that.